File: engine/Tag.js
import {Object as InkObject} from './Object';

export class Tag extends InkObject{
	constructor(tagText){
		super();
		this._text = tagText.toString() || '';
	}
	get text(){
		return this._text;
	}
	toString(){
		return "# " + this._text;
	}
}

File: ink_dart/lib/src/engine/tag.dart
import 'ink_object.dart';

/// Tag object representing metadata in the story.
///
/// Tags are marked with # in ink source and provide
/// metadata that can be queried during execution.
class Tag extends InkObject {
  final String text;

  Tag(this.text);

  @override
  String toString() => '# $text';

  @override
  InkObject copy() => Tag(text);
}
